web/src/scripts/containers/TabbedEditor.jsx

```diff
@@ -171,6 +176,43 @@ class TabbedEditor extends Component {
       this.setState({files})
       this.openMenu(MENU_FILE_SEARCH)
     },
+    focusNextGroup: (e) => {
+      const { dispatch } = this.props
+      dispatch(tabActions.focusAdjacentGroup(CONTENT_PANES.CENTER, (i) => i+1))
+    },
+    focusPrevGroup: (e) => {
+      const { dispatch } = this.props
+      dispatch(tabActions.focusAdjacentGroup(CONTENT_PANES.CENTER, (i) => i-1))
+    },
+    nextTab: (e) => {
+      const { dispatch } = this.props
+      dispatch(tabActions.focusAdjacentTab(CONTENT_PANES.CENTER, (i) => i+1))
+    },
+    prevTab: (e) => {
+      const { dispatch } = this.props
+      dispatch(tabActions.focusAdjacentTab(CONTENT_PANES.CENTER, (i) => i-1))
+    },
+    openInNewTab: (e) => {
+      const {decoDoc, dispatch} = this.props
+      if (!decoDoc) return
+      dispatch(tabActions.splitRight(CONTENT_PANES.CENTER, URIUtils.filePathToURI(decoDoc.id)))
+    }
+  }
+
+  constructor(props) {
+    super(props)
+    _.each(['1','2','3','4','5','6','7','8','9'], (num) => {
+      this.keyHandlers[`focusTab${num}`] = () => {
+        const { dispatch } = this.props
+        dispatch(tabActions.focusAdjacentTab(CONTENT_PANES.CENTER, (i) => Number(num) - 1, {nowrap: true}))
```

Comment:
Probably shouldn't be using the `Number` constructor here... maybe make the array numbers instead of strings.